common/src/stack/switch/pylib/switch/m7800.py

```diff
@@ -131,19 +145,23 @@ def wipe_ssh_keys(self):
 	def set_hostname(self, hostname):
 		self.subnet_manager = False
 		# this doesn't happen immediately..
-		time.sleep(.5)
+		while self.subnet_manager:
+			time.sleep(1)
+			self.subnet_manager = False
 		self.proc.say(f'hostname {hostname}')
 
 		for host in self._get_smnodes():
 			if host != hostname:
 				self.proc.say(f'no ib smnode {host}')
 
 		self.subnet_manager = True
-		time.sleep(.5)
+		while not self.subnet_manager:
```

Comment:
This while loop and the one above it appear to be able to get stuck forever if the switch decides to never listen to the update commands. Should these have some kind of timeout to fail after a reasonable amount of time?

Reply:
I put a cap on it in the latest rev.